### PR TITLE
style: Move text-align outside of the mako file.

### PR DESCRIPTION
**Pull request by canova**
I will need this refactoring before my next job. I didn't actually fix the FIXME's along the way. My other PR probably will cover these.   

---
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19576)

### CODE REVIEW

File: components/style/values/specified/text.rs [hunk 1]
```diff
@@ -349,3 +351,108 @@ impl Parse for TextDecorationLine {
         }
     }
 }
+
+/// Specified value of text-align property.
+#[cfg_attr(feature = "gecko", derive(MallocSizeOf))]
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub enum TextAlign {
+    /// Keyword value of text-align property.
+    Keyword(ComputedTextAlign),
```
Comment (Manishearth) [inline components/style/values/specified/text.rs]: Given that `ComputedTextAlign` is used in the specified value I'd prefer if it were in this file as `TextAlignKeyword` and `computed::TextAlign` is an alias

**Review by Manishearth — APPROVED** [review]

File: components/style/values/specified/text.rs [hunk 2]
```diff
@@ -349,3 +351,108 @@ impl Parse for TextDecorationLine {
...
+
+impl Parse for TextAlign {
+    fn parse<'i, 't>(_context: &ParserContext, input: &mut Parser<'i, 't>) -> Result<Self, ParseError<'i>> {
+        // MozCenterOrInherit cannot be parsed, only set directly on the elements
+        if let Ok(key) = input.try(ComputedTextAlign::parse) {
+            return Ok(TextAlign::Keyword(key));
+        }
+        #[cfg(feature = "gecko")]
+        {
+            if let Ok(_) = input.try(|i| i.expect_ident_matching("match-parent")) {
```
Comment (emilio) [inline components/style/values/specified/text.rs]: This doesn't need to use `if let`, just:

```rust
input.expect_ident_matching("match_parent")?;
return Ok(TextAlign::MatchParent);
```

**Review by emilio — APPROVED** [review]
r=me with @Manishearth's comment addressed and mine.

File: components/style/values/computed/text.rs [hunk 3]
```diff
@@ -101,3 +101,69 @@ impl ToCss for TextDecorationLine {
...
+        define_css_keyword_enum! { TextAlign:
+            $(
+                $string => $name,
+            )+
+        }
+
+        impl TextAlign {
+            /// Convert computed TextAlign into u32
+            pub fn to_u32(self) -> u32 {
+                match self {
```
Comment (emilio) [inline components/style/values/computed/text.rs]: This is super stupid btw (not your fault of course). I should just land the patch to `derive(Parse)` for keywords and use `self as u32`.

Comment (emilio) [reply components/style/values/computed/text.rs]: I'll actually land it, gimme a sec :)

Comment (emilio) [reply components/style/values/computed/text.rs]: #19578

**Comment by canaltinova** [comment]
@emilio Added a small commit to do that Parse derive. Could you look at it?

**Review by emilio — APPROVED** [review]
Arrg, still unfortunate we need the macros just for `from_u32`, but it's fine for now. We can maybe auto-generate that in the future.

File: components/style/values/specified/text.rs [hunk 4]
```diff
@@ -349,3 +350,168 @@ impl Parse for TextDecorationLine {
...
+        // MozCenterOrInherit cannot be parsed, only set directly on the elements
+        if let Ok(key) = input.try(TextAlignKeyword::parse) {
+            return Ok(TextAlign::Keyword(key));
+        }
+        #[cfg(feature = "gecko")]
+        {
+            input.expect_ident_matching("match_parent")?;
+            return Ok(TextAlign::MatchParent);
+        }
+        return Err(input.new_custom_error(StyleParseErrorKind::UnspecifiedError));
```
Comment (emilio) [inline components/style/values/specified/text.rs]: Travis has failed because this is unreachable on Gecko.

**Comment by canaltinova** [comment]
Actually let me create a `ToKeywordEnum` proc macro to implement `from_u32`. So we can get rid of that macro.

**Comment by canaltinova** [comment]
Can you look at it again to the new commit?

File: components/style_derive/to_keyword_enum.rs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    });
+
+    match_body = quote! {
+        #match_body
+        _ => None
+    };
+
+    // TODO(canaltinova): We can create a trait for that.
+    quote! {
+        impl #name {
```
Comment (emilio) [inline components/style_derive/to_keyword_enum.rs]: It'd be nice to make it a real trait if possible. also, it's kind of unfortunate that we fix this to `u32` enums, isn't it?

I think the macro is ok for now if there's only one use case.

**Comment by canaltinova** [comment]
Nvm, it doesn't worth the effort. Changed back to macro.
@bors-servo r=Manishearth,emilio

File: components/style/values/specified/text.rs [hunk 6]
```diff
@@ -349,3 +350,171 @@ impl Parse for TextDecorationLine {
...
+
+impl Parse for TextAlign {
+    fn parse<'i, 't>(_context: &ParserContext, input: &mut Parser<'i, 't>) -> Result<Self, ParseError<'i>> {
+        // MozCenterOrInherit cannot be parsed, only set directly on the elements
+        if let Ok(key) = input.try(TextAlignKeyword::parse) {
+            return Ok(TextAlign::Keyword(key));
+        }
+        #[cfg(feature = "gecko")]
+        {
+            input.expect_ident_matching("match_parent")?;
```
Comment (emilio) [inline components/style/values/specified/text.rs]: `match-parent` (s/_/-/)

**Review by emilio — CHANGES_REQUESTED** [review]

**Comment by emilio** [comment]
(Feel free to land with that fixed and the rest of the patch sanity-checked, thanks!)

**Comment by canaltinova** [comment]
Oops, thanks!
@bors-servo r=Manishearth,emilio

**Pull request merged**